argo_workflow_tools/dsl/utils/utils.py

```diff
@@ -83,6 +83,11 @@ def sanitize_name(name: str, snake_case=False) -> str:
     return name.replace("_", "-")
 
 
+def generate_template_name_from_func(func: Callable, snake_case=False) -> str:
+    sanitized = sanitize_name(func.__name__, snake_case)
+    return f'{sanitized}-{func.__hash__()}'
```

Comment:
This is main change # 2 in this PR

Reply:
my preference - 
1) can we use the workflow template as context instead? i.e. classic namespace. that would be the ideal solution. the "root" workflow of course doesn't need to do this. which will keep the code both clean and backward compatible. 
2) if not, consider using uuid_short() from dsl.utils as the hash names of these templates can be long as hell and not very fun to work with. we must remember we must be able to keep working ith the workflow in the UI.